modules/tensor_mechanics/src/materials/ComputeAdmissibleState.C

```diff
@@ -0,0 +1,281 @@
+              stress_min(i, j) = _stress[_qp](i, j);
+          }
+        }
+      }
+    }
+
+    // now check convergence in the stress:
+    // once the change in stress is within tolerance after each recompute material
+    // consider the stress to be converged
+    RankTwoTensor delta_stress(stress_max - stress_min);
+    l2norm_delta_stress = delta_stress.L2norm();
+    if (counter == 0)
+      first_l2norm_delta_stress = l2norm_delta_stress;
+
+    if (_output_iteration_info == true)
+    {
+      _console << "stress iteration number = " << counter << "\n"
+               << " relative l2 norm delta stress = "
+               << (0 == first_l2norm_delta_stress ? 0
+                                                  : l2norm_delta_stress / first_l2norm_delta_stress)
+               << "\n"
+               << " stress convergence relative tolerance = " << _relative_tolerance << "\n"
+               << " absolute l2 norm delta stress = " << l2norm_delta_stress << "\n"
+               << " stress converengen absolute tolerance = " << _absolute_tolerance << std::endl;
+    }
+    ++counter;
+  } while (counter < _max_its && l2norm_delta_stress > _absolute_tolerance &&
+           (l2norm_delta_stress / first_l2norm_delta_stress) > _relative_tolerance &&
+           _num_models != 1);
+
+  if (counter == _max_its && l2norm_delta_stress > _absolute_tolerance &&
+      (l2norm_delta_stress / first_l2norm_delta_stress) > _relative_tolerance)
+    mooseError("Max stress iteration hit during ComputeAdmissibleState solve!");
+
+  combined_inelastic_strain_increment.zero();
+  for (unsigned i_rmm = 0; i_rmm < _num_models; ++i_rmm)
+    combined_inelastic_strain_increment +=
+        _inelastic_weights[i_rmm] * inelastic_strain_increment[i_rmm];
+
+  if (_tangent_operator_type == TangentOperatorEnum::elastic)
+    _Jacobian_mult[_qp] = _elasticity_tensor[_qp];
+  else
+  {
+    const RankFourTensor E_inv = _elasticity_tensor[_qp].invSymm();
+    _Jacobian_mult[_qp] = _cto[0];
+    for (unsigned i_rmm = 1; i_rmm < _num_models; ++i_rmm)
+      _Jacobian_mult[_qp] = _cto[i_rmm] * E_inv * _Jacobian_mult[_qp];
+  }
+}
+
+void
+ComputeAdmissibleState::updateQpStateSingleModel(
+    RankTwoTensor & elastic_strain_increment, RankTwoTensor & combined_inelastic_strain_increment)
+{
+  _models[0]->setQp(_qp);
+
+  elastic_strain_increment = _strain_increment[_qp];
+
+  _stress[_qp] = _stress_old[_qp] + _elasticity_tensor[_qp] * elastic_strain_increment;
+
+  _models[0]->updateState(elastic_strain_increment,
+                          combined_inelastic_strain_increment,
+                          _rotation_increment[_qp],
+                          _stress[_qp],
+                          _stress_old[_qp],
+                          _elasticity_tensor[_qp],
+                          _elastic_strain_old[_qp],
+                          _tangent_operator_type == TangentOperatorEnum::nonlinear,
+                          _Jacobian_mult[_qp]);
```

Comment:
I can, but it's suboptimal because it requires copying the RankFourTensor in some cases.  Can you set these in your RadialReturnStressUpdate?  You have the interface

```
updateState(... , elasticity_tensor, ... compute_full_tangent, tangent_operator)
```
so can you just put
```
tangent_operator = elasticity_tensor
```
at the end of your RadialReturnStressUpdate (or whatever class derives from StateUpdateBase)?  That's how it's supposed to work (see around line 170 of RadialReturnStressUpdate)

a